example-app/src/index.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ChangeEvent, useEffect, useState } from "react";
+import ReactDOM from "react-dom";
+import * as helpers from "./helpers"
+
+// Simple app handling configuration and using helpers.
+const AppComponent = () => {
+  const [tokenServiceEnv, setTokenServiceEnv] = useState(localStorage.getItem("tokenServiceEnv") || "dev");
+  const [portalUrl, setPortalUrl] = useState(localStorage.getItem("portalUrl") || "https://app.rigse.docker");
+  const [oauthClientName, setOauthClientName] = useState(localStorage.getItem("oauthClientName") || "localhost-public");
+  const [firebaseAppName, setFirebaseAppName] = useState(localStorage.getItem("firebaseApp") || "token-service");
+  const [portalAccessToken, setPortalAccessToken] = useState("");
+  const [firebaseJwt, setFirebaseJwt] = useState("");
+  const [fileContent, setFileContent] = useState("Lorem ipsum");
+  const [filePublicUrl, setFilePublicUrl] = useState("");
+
+  useEffect(() => {
+    setPortalAccessToken(helpers.readPortalAccessToken());
+  }, []);
+
+  const changeText = (prop: string, event: ChangeEvent<HTMLInputElement>) => {
+    if (prop === "portalUrl") {
+      setPortalUrl(event.target.value);
+    } else if (prop === "oauthClientName") {
+      setOauthClientName(event.target.value);
+    } else if (prop === "firebaseAppName") {
+      setFirebaseAppName(event.target.value);
+    } else if (prop === "fileContent") {
+      setFileContent(event.target.value);
+    } else if (prop === "tokenServiceEnv") {
+      setTokenServiceEnv(event.target.value);
+    }
+    localStorage.setItem(prop, event.target.value);
+  };
+
+  const handleAuthorizeInPortal = () => {
+    helpers.authorizeInPortal(portalUrl, oauthClientName);
+  };
+
+  const handleGetFirebaseJwt = () => {
+    helpers.getFirebaseJwt(portalUrl, portalAccessToken, firebaseAppName).then(token => setFirebaseJwt(token));
+  };
+
+  const handleUploadFile = async () => {
+    setFilePublicUrl(await helpers.uploadFileUsingFirebaseJTW(fileContent, firebaseJwt, tokenServiceEnv as "dev" | "staging"));
+  };
+
+  return (
+    <div>
+      <div className="section">
+        <h3>Portal Setup</h3>
+        <p className="hint">
+          Token Service requires FirebaseJWT that can be obtained from Portal. Provide Portal configuration, click
+          "Authorize in Portal" (note that it'll reload this page) and finally "Get FirebaseJWT". When FirebaseJWT
```

Comment:
You might update this description to explain that it will require a login at the portal if your user is not currently logged in.
I'd also suggest changing "finally" to "and then click".